MyApplication/app/src/main/java/com/colinbradley/syncadapterlab/SyncAdapter.java

```diff
@@ -0,0 +1,88 @@
+package com.colinbradley.syncadapterlab;
+
+import android.accounts.Account;
+import android.content.AbstractThreadedSyncAdapter;
+import android.content.ContentProviderClient;
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.SyncResult;
+import android.database.Cursor;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by colinbradley on 12/8/16.
+ */
+
+public class SyncAdapter extends AbstractThreadedSyncAdapter{
+    public static final String TAG = "SyncAdapter";
+
+    ContentResolver mContentResolver;
+    Context mContext;
+
+    public SyncAdapter(Context context, boolean autoInitialize) {
+        super(context, autoInitialize);
+
+        mContentResolver = context.getContentResolver();
+        mContext = context;
```

Comment:
Don't save a reference to context if you don't have to - and in this case you don't. Objects that extend `Context`, like `Activity` or `Service`, are big and take up a lot of memory, so we don't want to prevent them from being garbage collected by saving a reference to them that we don't actually need.